modules/tinymce/src/core/main/ts/util/Quirks.ts

```diff
@@ -249,6 +249,20 @@ const Quirks = (editor: Editor): Quirks => {
     });
   };
 
+  /**
+   * Fixes a Gecko a selection bug where if there is a floating image
+   * more details here: https://bugzilla.mozilla.org/show_bug.cgi?id=1959606
+   */
+  const fixFirefoxImageSelection = () => {
+    editor.on('selectionchange', () => {
+      const rng = editor.selection.getRng();
+      if (rng.collapsed && rng.startContainer.nodeName === 'IMG') {
+        editor.selection.select(rng.startContainer);
+        editor.selection.collapse(true);
```

Comment:
This seems a bit odd? Why is this needed to me that would mean the caret gets placed before the image not on the image if you click on it.
Maybe this needs to be more advanced since if the caret can be moved into the image on multiple scenarios like.
When you click before an image, when you click on an image and when you click after an image. 
I tinkered a bit with a fiddle of that: https://fiddle.tiny.cloud/c8KXYW1BG2/0
We want to detect when the selection will become wrong on the default action of `mousedown` since if we detect it early we don't get a flicker where the caret is first somewhere incorrect then snaps into the correct position.
Also on `selectionchange` we don't have any X, Y cord to inspect to figure out if the click was before/after or on the image.

Reply:
changed thanks